sonar-php-plugin/src/main/java/org/sonar/plugins/php/phpunit/PhpUnitTestFileReport.java

```diff
@@ -0,0 +1,152 @@
+import org.sonar.plugins.php.api.Php;
+import org.sonar.plugins.php.phpunit.xml.TestCase;
+
+/**
+ * The PhpUnitTestFileReport contains all the results of test cases appearing in a given test file.
+ * The reason why the report is file-based (as opposed to class-based) is that the SonarQube measures
+ * are stored per file.
+ */
+public class PhpUnitTestFileReport {
+
+  public PhpUnitTestFileReport(String file, double testDuration) {
+    this.file = file;
+    this.testDuration = testDuration;
+  }
+  private static final Logger LOGGER = LoggerFactory.getLogger(PhpUnitTestResultImporter.class);
+  private int errors = 0;
+  private int failures = 0;
+  private String file;
+  private int skipped = 0;
+  private int tests = 0;
+
+  private double testDuration = 0;
+
+  public void saveTestMeasures(SensorContext context) {
+    InputFile unitTestFile = getUnitTestInputFile(context.fileSystem());
+    if (unitTestFile != null) {
+      context.<Integer>newMeasure().on(unitTestFile).withValue(skipped).forMetric(CoreMetrics.SKIPPED_TESTS).save();
+
+      context.<Long>newMeasure().on(unitTestFile).withValue((long) testDurationSeconds()).forMetric(CoreMetrics.TEST_EXECUTION_TIME).save();
+      context.<Integer>newMeasure().on(unitTestFile).withValue((int) liveTests()).forMetric(CoreMetrics.TESTS).save();
+      context.<Integer>newMeasure().on(unitTestFile).withValue(errors).forMetric(CoreMetrics.TEST_ERRORS).save();
+      context.<Integer>newMeasure().on(unitTestFile).withValue(failures).forMetric(CoreMetrics.TEST_FAILURES).save();
+      if (liveTests() > 0) {
+        context.<Double>newMeasure().on(unitTestFile).withValue(ParsingUtils.scaleValue(successPercentage())).forMetric(CoreMetrics.TEST_SUCCESS_DENSITY).save();
+      }
+
+    } else {
+      LOGGER.debug("Following file is not located in the test folder specified in the Sonar configuration: " + file
+        + ". The test results won't be reported in Sonar.");
+    }
+  }
+
+  private double liveTests() {
+    return (double) tests - skipped;
+  }
+
+  private double successPercentage() {
+    double passedTests = liveTests() - errors - failures;
+    return passedTests * 100d / liveTests();
+  }
+
+  private double testDurationSeconds() {
+    return testDuration * 1000d;
+  }
+
+  private InputFile getUnitTestInputFile(FileSystem fileSystem) {
+    FilePredicates predicates = fileSystem.predicates();
+    return fileSystem.inputFile(predicates.and(
+      predicates.hasPath(file),
+      predicates.hasType(InputFile.Type.TEST),
+      predicates.hasLanguage(Php.KEY)));
+  }
+
+  public void addTestCase(TestCase testCase) {
+    if (testCase.getStatus() == TestCase.Status.SKIPPED) {
+      this.skipped++;
+    } else if (testCase.getStatus() == TestCase.Status.FAILURE) {
+      this.failures++;
+    } else if (testCase.getStatus() == TestCase.Status.ERROR) {
+      this.errors++;
+    }
+    this.tests++;
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
Yes! Better than a dozen getters and it does not bring people to misappropriate methods created only for test purposes.